111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,239 @@
+
+We will use following commands:
+
+```shell
+buildah build --file=/dockerfile/Dockerfile --tag=<IMAGE> --storage-driver=vfs <ADDITIONAL_BUILD_OPTS>
+buildah push --storage-driver=vfs --digestfile=/tmp/digest <ADDITIONAL_PUSH_OPTS> <IMAGE>
+buildah images --digests --filter=digest=sha256:$(cat /tmp/digest) --format='{{.Name}}@{{.Digest}}' > /dev/termination-log
+```
+- `<IMAGE>` is placeholder for user desired name of the image (with registry, repository, and possibly tag)
+- `<ADDITIONAL_BUILD_OPTS>` is placeholder for user desired additional options for the `buildah build` command
+- `<ADDITIONAL_PUSH_OPTS>` is placeholder for user desired additional options for the `buildah push` command
+
+We need to take these three steps in order to get the SHA of the image together with correct repository.
+This will work in all scenarios, including when the user does not specify a tag and the default (`latest`) is used.
+
+Together with this feature, and upcoming v1 API, we will do following changes inside the `DockerOutput` model:
+
+1. we will deprecate `additionalKanikoOptions` field as the name of it is connected directly to Kaniko and more generic name can be used (this field will be removed as part of the v1 API).
+2. we will add two new fields - `additionalBuildOptions` and `additionalPushOptions` - to cover additional options for both phases
+
+The `additionalKanikoOptions` will still be used with Kaniko until the `UseConnectBuildWithBuildah` feature gate moves to GA or until it is removed in the CRD v1 API, whatever comes first.
+An automated warning about deprecation will be added inside the `.status` section of the `KafkaConnect` CR in case it is used as for all other deprecated fields.
+Instead of `additionalKanikoOptions`, users should use `additionalBuildOptions`, but not `additionalPushOptions`.
+We will not split the Kaniko's additional options to two groups, and we will keep everything configured in one.
+That's because Kaniko doesn't have the build and push phases split, but everything is done in one command.
+The difference between the build and push options in Kaniko is that the `-pull` suffix is applied for options used only for the build phase.
+
+Buildah, in the other hand, has these phases split and has its own set of options for both.
+In order to allow specifying different options and their values for both of these phases, we will add two fields.
+Additionally, if user will use Kaniko and specify both `additionalKanikoOptions` and `additionalBuildOptions`, `additionalBuildOptions` takes precedence.
+
+For Buildah, we will check only the `additionalBuildOptions` and `additionalPushOptions` fields.
+In case that user will specify `additionalKanikoOptions` with Buildah enabled feature gate, two conditions with warnings will be added into `.status` section of `KafkaConnect` CR - one about deprecation and one about `additionalKanikoOptions` being ignored, as Buildah is used.
```

Comment:
I don't think this needs any additional status warning. They will find out pretty quickly.